docs/developer/designs/dra-claim-sharing/dra-claim-sharing.md

```diff
@@ -0,0 +1,61 @@
+Even though the claims are namespaced, and can't be shared across namespaces - KAI (by design) doesn't enforce queue-namespace exclusivity, meaning multiple queues can use the same namespace. This means that, theoretically, pods from multiple queues can access the same GPU resource claim, which poses a resource accounting problem: which queue should bear the "cost" of these resources?
+
+## Proposed Solution
+
+In essence, the proposed solution is: 
+1. Link each resource claim to a queue, which will account for it, no matter who else uses it
+2. Allow queues to opt-out of sharing their resources with others
+3. Don't add these requirements for resources that are not accounted in queues
+
+### Details
+
+#### Linking Resource Claims to Queues
+
+The easy way to link resource claims to queues is to use a queue label, like we do for podgroups. A webhook can be used to make this immutable (which we should probably do for podgroups as well, by the way). However, this might make things more difficult for users, and even clash with DRA objects which are created by third parties, or simple examples.  
+We could do an implicit linking of objects to queues: DRA objects are bound by the binder, according to the BindRequest spec which is generated by the scheduler, and includes the DRA allocation result. The scheduler can define the queue owner of the claim as the queue of the first pod to bind this ResourceClaim. The binder can then be responsible for adding the label to the claim, to be used by other services. The internal BindRequestInfo in the scheduler can be used to track internally the queue owner of the bind request.
+
+#### Non-accounted resources
+
+It stands to reason that not all DRA resources will be managed by queues: for example, some resources might represent resources which are not in contention and are not meant to be divided by KAI. We can define, either in scheduler shard configuration, or on the device classes, which resources are meant to be accounted by KAI.
+
+### Complications
+
+#### Resource Locking
+
+Imagine queue A creates a claim and lets queue B share it. Queue B allocates a non-preemptible pod on this claim. Queue A's pod is evicted or finishes running, but queue B's non-preemptible pod keeps running, keeping the claim bound and the resources allocated.
+
+Possible solutions:
+1. Don't allow foreign queues' non-preemptible pods to share claims
+    - What about pods which are protected by min runtime?
+    - Could probably be the easiest to implement at first, until we get familiar with more use cases
+2. Allow queues to evict claims, even if it will evict foreign pods
+    - Need to think how to implement this - maybe an annotation on the claim?
```

Comment:
Maybe a third option would be to "transfer ownership" of the claim where the foreign queue assumes the claim's cost when the original queue has not pods on it.